server/services/licenceService.ts

```diff
@@ -459,6 +465,60 @@ export default class LicenceService {
     return this.licenceApiClient.notifyComsToPromptEmailCreation(emailGroups)
   }
 
+  async getTimelineEvents(licence: Licence, user: User): Promise<TimelineEvent[]> {
+    const licences: Licence[] = []
+    let thisLicence: Licence = licence
+    licences.push(thisLicence)
+
+    // Get the trail of variations back to the original licence
+    while (thisLicence?.isVariation) {
+      // eslint-disable-next-line no-await-in-loop
+      thisLicence = await this.licenceApiClient.getLicenceById(`${thisLicence?.variationOf}`, user)
+      if (thisLicence) {
+        licences.push(thisLicence)
+      }
+    }
+
+    const timelineEvents = this.convertLicencesToTimelineEvents(licences)
+
+    logger.info(`Timeline Events = ${JSON.stringify(timelineEvents)}`)
+
+    return timelineEvents
+  }
+
+  private convertLicencesToTimelineEvents(licences: Licence[]): TimelineEvent[] {
+    return licences.map(licence => {
+      const { title, eventType } = this.getTimelineEventType(licence.variationOf, licence.statusCode)
+      return new TimelineEvent(
+        eventType,
+        title,
+        licence.statusCode,
+        licence.createdByFullName,
+        licence.id,
+        licence.dateLastUpdated
+      )
+    })
+  }
+
+  private getTimelineEventType(varyOf: number, status: string): { eventType: TimelineEventType; title: string } {
+    switch (status) {
+      case LicenceStatus.VARIATION_IN_PROGRESS:
+      case LicenceStatus.VARIATION_REJECTED:
+      case LicenceStatus.VARIATION_SUBMITTED:
+        return { eventType: TimelineEventType.VARIATION_IN_PROGRESS, title: 'Variation in progress' }
+
+      case LicenceStatus.VARIATION_APPROVED:
+        return { eventType: TimelineEventType.VARIATION, title: 'Licence varied' }
+
+      case LicenceStatus.ACTIVE:
+      case LicenceStatus.INACTIVE:
+      default:
+        return varyOf
+          ? { eventType: TimelineEventType.VARIATION, title: 'Licence varied' }
+          : { eventType: TimelineEventType.CREATION, title: 'Licence created' }
+    }
+  }
+
```

Comment:
Might I suggest this is moved to a new service class called `TimelineService` which consumes `LicenceService`? Feels like something which could grow into a plethora of business logic on its own, and `LicenceService` is already big

Reply:
Feels like a good idea, though maybe a follow-on refactor piece once its in?